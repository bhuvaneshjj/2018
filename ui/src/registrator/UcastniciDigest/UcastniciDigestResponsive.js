import PropTypes from 'prop-types';
import Dimensions from 'react-dimensions';
import UcastniciDigest from './UcastniciDigest';

const UcastniciDigestResponsive = Dimensions({
  getHeight: () => window.innerHeight - 130,
  getWidth: () => window.innerWidth - 100
})(UcastniciDigest);

// ucastniciDigest are simple pass-through from UcastniciDigestContainer.
UcastniciDigestResponsive.propTypes = {
  sortColumn: PropTypes.string,
  sortDir: PropTypes.string,
  ucastniciDigest: PropTypes.array.isRequired,
  fetchUcastnici: PropTypes.func.isRequired,
  onSortDirChange: PropTypes.func.isRequired
};

export default UcastniciDigestResponsive;
