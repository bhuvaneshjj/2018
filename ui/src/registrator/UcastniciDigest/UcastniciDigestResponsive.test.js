import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import UcastniciDigestResponsive from './UcastniciDigestResponsive';

const ucastniciDigest = [];

it('renders', () => {
  global.window = { innerHeight: 700, innerWidth: 700 };
  const wrapper = mount(
    <div height={500} width={500}>
      <UcastniciDigestResponsive
        ucastniciDigest={ucastniciDigest}
        fetchUcastnici={jest.fn()}
        onSortDirChange={jest.fn()}
      />
    </div>
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});
