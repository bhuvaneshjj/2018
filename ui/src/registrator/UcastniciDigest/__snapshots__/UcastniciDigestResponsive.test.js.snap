// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  height={500}
  width={500}
>
  <DimensionsHOC
    fetchUcastnici={[Function]}
    onSortDirChange={[Function]}
    ucastniciDigest={Array []}
  >
    <div
      className={null}
      style={
        Object {
          "border": 0,
          "height": "100%",
          "padding": 0,
          "width": "100%",
        }
      }
    >
      <UcastniciDigest
        containerHeight={638}
        containerWidth={924}
        fetchUcastnici={[Function]}
        onSortDirChange={[Function]}
        ucastniciDigest={Array []}
        updateDimensions={[Function]}
      >
        <div
          className="Ucastnici"
        >
          žádný účastník
        </div>
      </UcastniciDigest>
    </div>
  </DimensionsHOC>
</div>
`;
